### PR TITLE
Federated replica set design doc

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @quinton-hoole @kubernetes/sig-cluster-federation

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34090)

### CODE REVIEW

**Comment by mwielgus** [comment]
Updated version of #33579.

**Comment by quinton-hoole** [comment]
@mwielgus What's changed relative to the original design doc that I've already reviewed?

**Comment by mwielgus** [comment]
goroutines and scheduling algorithm description

**Comment by mwielgus** [comment]
@quinton-hoole  ping

File: docs/design/federated-replicasets.md [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+   +  Some replica set inside changed its status
+   +  Some pods inside cluster changed their status
+   +  Some cluster is added or deleted.
+> FRS stops ToBeChecked if is in desired configuration (or is stable enough).
+
+## (RE)Scheduling algorithm
+
+To calculate the (re)scheduling moves for a given FRS:
+
+1. For each cluster FRSC calculates the number of replicas that are placed
```
Comment (quinton-hoole) [inline docs/design/federated-replicasets.md]: nit: Add comma after 'cluster'.

File: docs/design/federated-replicasets.md [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+> FRS stops ToBeChecked if is in desired configuration (or is stable enough).
+
+## (RE)Scheduling algorithm
+
+To calculate the (re)scheduling moves for a given FRS:
+
+1. For each cluster FRSC calculates the number of replicas that are placed 
+(not necessary up and running) in the cluster and the number of replicas that 
+failed to be scheduled. Cluster capacity is the difference between the
+the placed and failed to be scheduled.
```
Comment (quinton-hoole) [inline docs/design/federated-replicasets.md]: Sorry, I don't understand this sentence.  Perhaps I'm being dumb.  How can cluster capacity be defined this way? e.g. If I try to schedule 10, and 9 schedule, leaving 1 unscheduled, it seems that the remaining capacity should be zero, not 9-1=8.  Does this sentence need rewording?

**Comment by quinton-hoole** [comment]
@mwielgus nits then LGTM.  Feel free to apply the label once addressed.

**Pull request merged**